R/data.table.R

```diff
@@ -1335,7 +1335,9 @@ replace_dot_alias = function(e) {
     }
 
     if (is.data.table(jval)) {
-      setattr(jval, 'class', class(x)) # fix for #64
+      # should set the parent class only when jval is a plain data.table #4324
+      if (setequal(class(jval), c('data.table', 'data.frame'))) 
```

Comment:
why `setequal` and not `identical`?
```r
setequal(c("a","b"), c("b","a"))
```
do we want to support `c("data.frame","data.table")` as well?

Reply:
You are right. Fixed.